OpenRA.Mods.Common/Widgets/Logic/ModBrowserLogic.cs

```diff
@@ -140,18 +147,46 @@ void RebuildModList()
 			}
 		}
 
-		void SelectMod(ModMetadata mod)
+		void SelectMod(ModMetadata mod, bool? isInstalled)
 		{
+			if (isInstalled == null)
+				isInstalled = IsModInstalled(mod);
+
 			selectedMod = mod;
 			selectedAuthor = "By " + (mod.Author ?? "unknown author");
 			selectedDescription = (mod.Description ?? "").Replace("\\n", "\n");
 			var selectedIndex = Array.IndexOf(allMods, mod);
 			if (selectedIndex - modOffset > 4)
 				modOffset = selectedIndex - 4;
+
+			loadButton.Text = isInstalled.Value ? "Load Mod" : "Install Assets";
 		}
 
-		void LoadMod(ModMetadata mod)
+		void LoadMod(ModMetadata mod, bool? isInstalled)
 		{
+			if (isInstalled == null)
+				isInstalled = IsModInstalled(mod);
+
+			if (!isInstalled.Value)
+			{
+				var installData = mod.AssetInformation;
+
+				var widgetArgs = new WidgetArgs
+				{
+					{ "continueLoading", () =>
+						Game.RunAfterTick(() => Game.InitializeMod(Game.Settings.Game.Mod, new Arguments())) },
+					{ "mirrorListUrl", installData.PackageMirrorList },
+					{ "modId", mod.Id }
+				};
+
+				modChooserPanel.Get<BackgroundWidget>("bg").Visible = false;
```

Comment:
You can / should adjust the mod chooser to open these as a Window so that you don't need to manually hide/show them.